### PR TITLE
[js] Add a check for Grid CDP endpoint

### LINKED ISSUE #9779: Javascript client driver.createCDPConnection promise never resolved with usingServer in webdriver builder
## 🐛 Bug Report
Selenium jar version: `4.0.0-beta-4`
Javascript client version: `4.0.0-beta.4`
OS version: `MacOS Catalina`
Javascript client promise is never is resolved if we create driver as follows.
```
  let driver = new webdriver.Builder()
  .forBrowser('chrome')
  .usingServer('http://localhost:4444')
  .build();
```
and use CDP as follows.
```
const cdpConnection = await driver.createCDPConnection('page');
```
The same script works if I remove `usingServer` and use `chromedriver` directly.
## To Reproduce
Run following javascript code.
```
const webdriver = require('selenium-webdriver');

const run = async () => {
  let driver = new webdriver.Builder()
  .forBrowser('chrome')
  .usingServer('http://localhost:4444')
  .build();
  try {
    const cdpConnection = await driver.createCDPConnection('page');
    await driver.onLogEvent(cdpConnection, (event) => {
      console.log(event);
    });
    await driver.get('http://facebook.com');
  } finally {
    setTimeout(() => { driver.quit() }, 10000);
  }
}

run();
```
Code will not make it past `const cdpConnection = await driver.createCDPConnection('page');` line.
Detailed steps to reproduce the behavior:
Run Selenium jar
```
java -jar selenium.jar standalone
```
Then run this code (same as earlier)
```
const webdriver = require('selenium-webdriver');

const run = async () => {
  let driver = new webdriver.Builder()
  .forBrowser('chrome')
  .usingServer('http://localhost:4444')
  .build();
  try {
    const cdpConnection = await driver.createCDPConnection('page');
    await driver.onLogEvent(cdpConnection, (event) => {
      console.log(event);
    });
    await driver.get('http://facebook.com');
  } finally {
    setTimeout(() => { driver.quit() }, 10000);
  }
}

run();
```
But if we do not use `.usingServer('http://localhost:4444')` it works fine provided `chromedriver` is in the `PATH`.
i.e. following code works.
```
const webdriver = require('selenium-webdriver');

const run = async () => {
  let driver = new webdriver.Builder()
  .forBrowser('chrome')
  // .usingServer('http://localhost:4444')
  .build();
  try {
    const cdpConnection = await driver.createCDPConnection('page');
    await driver.onLogEvent(cdpConnection, (event) => {
      console.log(event);
    });
    await driver.get('http://facebook.com');
  } finally {
    setTimeout(() => { driver.quit() }, 10000);
  }
}

run();
```
## Expected behavior
Behavior should be the same with `usingServer` and without `usingServer`. I noticed that in client code it tries to make [HTTP request](https://github.com/SeleniumHQ/selenium/blob/40936c627d0da0fa502c61ae14a3305c7f20c951/javascript/node/selenium-webdriver/lib/webdriver.js#L1239) the WebSocket URL returned in `se:cdp` capability as [debuggerUrl](https://github.com/SeleniumHQ/selenium/blob/40936c627d0da0fa502c61ae14a3305c7f20c951/javascript/node/selenium-webdriver/lib/webdriver.js#L1200). I think the issue lies in that code.
Selenium jar returns `se:cdp` as `ws://192.168.42.127:4444/session/14a41fd2f6ec134a3683ead66eb291ae/se/cdp`
and client code modifies it to `http://ws://192.168.42.127:4444/session/14a41fd2f6ec134a3683ead66eb291ae/se/cdp`
Besides, an http request to `http://192.168.42.127:4444/session/14a41fd2f6ec134a3683ead66eb291ae/se/cdp` or `http://192.168.42.127:4444/session/14a41fd2f6ec134a3683ead66eb291ae/se/cdp/json` also doesn't work as Selenium jar throws following:
```
"error":"unknown command","message":"unknown command: unknown command: session/14a41fd2f6ec134a3683ead66eb291ae/se/cdp"
```
## Test script or set of commands reproducing this issue
Run Selenium 4.0.0-beta-4: `java -jar selenium.jar standalone`
Create a selenium node.js project: `npm init` 
Install javascript library selenium-webdriver-4.0.0-beta-4: `npm install selenium-webdriver`
Create node.js script `test.js`:
```
const webdriver = require('selenium-webdriver');

const run = async () => {
  let driver = new webdriver.Builder()
  .forBrowser('chrome')
  .usingServer('http://localhost:4444')
  .build();
  try {
    const cdpConnection = await driver.createCDPConnection('page');
    await driver.onLogEvent(cdpConnection, (event) => {
      console.log(event);
    });
    await driver.get('http://facebook.com');
  } finally {
    setTimeout(() => { driver.quit() }, 10000);
  }
}

run();
```
Run the node.js script: `node test.js`
## Environment

OS: `OS X Catalina`
Browser: `Chrome`
Browser version: `92.0.4515.159 (Official Build) (x86_64)`
Browser Driver version: `ChromeDriver 92.0.4515.107 (87a818b10553a07434ea9e2b6dccf3cbe7895134-refs/branch-heads/4515@{#1634})`
Language Bindings version: `Javascript 4.0.0-beta-4`
Selenium Grid version (if applicable): `4.0.0-beta-4`

### PR DESCRIPTION
**Pull request by pujagani**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

Add a check for Grid CDP endpoint and accordingly return the URL for websocket connection. Fixes #9779.

### Motivation and Context

When a remote web driver is built and a create CDP connection request is made, it just waits for an indefinite period of time. This happens because the client tries to send a request to the endpoint of this format "http://ws://session/{session-id}/se/cdp" to the Grid server. Since the Grid does not recognize this, the client does not receive any response. This is fixed by checking if the endpoint received from "se:cdp" capability is a Grid CDP endpoint and accordingly it is handled. The client then uses this endpoint to make a websocket connection with the Grid. 

### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [X] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: javascript/node/selenium-webdriver/lib/webdriver.js [hunk 1]
```diff
@@ -1229,6 +1229,14 @@ class WebDriver {
     if (target && cdpTargets.indexOf(target.toLowerCase()) === -1) {
       throw new error.InvalidArgumentError('invalid target value')
     }
+
+    if (debuggerAddress.match(/\/se\/cdp/)) {
```
Comment (AutomatedTester) [inline javascript/node/selenium-webdriver/lib/webdriver.js:1233]: if we are returning the `ws` here when passing in `se:cdp` we can just return it straight or do we have to make a request to get the `ws` address?

Comment (pujagani) [reply javascript/node/selenium-webdriver/lib/webdriver.js:1233]: We do not need to make a request to get the `ws` address. When the request is made here, it asks the Grid to upgrade to a socket and then the Grid makes the request to `/json/version` to get the address and uses that.

File: javascript/node/selenium-webdriver/lib/webdriver.js [hunk 2]
```diff
@@ -1229,6 +1229,14 @@ class WebDriver {
...
+      if (debuggerAddress.match("ws:\/\/", "http:\/\/")) {
```
Comment (AutomatedTester) [inline javascript/node/selenium-webdriver/lib/webdriver.js:1234]: What would happen here if there is a `wss`?

File: javascript/node/selenium-webdriver/lib/webdriver.js (hunk 1 above)
Comment (AutomatedTester) [reply javascript/node/selenium-webdriver/lib/webdriver.js:1233]: ok! 👍

File: javascript/node/selenium-webdriver/lib/webdriver.js (hunk 2 above)
Comment (pujagani) [reply javascript/node/selenium-webdriver/lib/webdriver.js:1234]: Thank you for bringing that to notice. Have added an additional check for the same and tested it locally with the Grid.

**Pull request merged**